lms/djangoapps/instructor/tests/test_api.py

```diff
@@ -4494,6 +4495,180 @@ def test_show_student_extensions(self):
                 'title': (f'Due date extensions for {self.user1.profile.name} ({self.user1.username})')}
 
 
+class TestChangeDueDateV2(SharedModuleStoreTestCase, LoginEnrollmentTestCase):
+    """
+    Tests for the V2 due date extension API endpoint.
+    """
+    MODULESTORE = TEST_DATA_SPLIT_MODULESTORE
+
+    @classmethod
+    def setUpClass(cls):
+        super().setUpClass()
+        cls.course = CourseFactory.create()
+        cls.due = datetime.datetime(2010, 5, 12, 2, 42, tzinfo=UTC)
+
+        with cls.store.bulk_operations(cls.course.id, emit_signals=False):
+            cls.week1 = BlockFactory.create(due=cls.due)
+            cls.week2 = BlockFactory.create(due=cls.due)
+            cls.week3 = BlockFactory.create()  # No due date
+            cls.course.children = [
+                str(cls.week1.location),
+                str(cls.week2.location),
+                str(cls.week3.location)
+            ]
+            cls.homework = BlockFactory.create(
+                parent_location=cls.week1.location,
+                due=cls.due
+            )
+            cls.week1.children = [str(cls.homework.location)]
+
+    def setUp(self):
+        """
+        Fixtures.
+        """
+        super().setUp()
+
+        self.user1 = UserFactory.create()
+        self.user2 = UserFactory.create()
+
+        # Create StudentModule objects for tracking student progress
+        StudentModule(
+            state='{}',
+            student_id=self.user1.id,
+            course_id=self.course.id,
+            module_state_key=self.week1.location).save()
+        StudentModule(
+            state='{}',
+            student_id=self.user1.id,
+            course_id=self.course.id,
+            module_state_key=self.homework.location).save()
+        StudentModule(
+            state='{}',
+            student_id=self.user2.id,
+            course_id=self.course.id,
+            module_state_key=self.week1.location).save()
+        StudentModule(
+            state='{}',
+            student_id=self.user2.id,
+            course_id=self.course.id,
+            module_state_key=self.homework.location).save()
+
+        CourseEnrollmentFactory.create(user=self.user1, course_id=self.course.id)
+        CourseEnrollmentFactory.create(user=self.user2, course_id=self.course.id)
+        self.instructor = InstructorFactory(course_key=self.course.id)
+        self.client.login(username=self.instructor.username, password=self.TEST_PASSWORD)
+        extract_dates(None, self.course.id)
+
+    def test_change_due_date_v2_success(self):
+        """Test successful due date change using V2 API endpoint"""
+        url = reverse('instructor_api_v2:change_due_date', kwargs={'course_id': str(self.course.id)})
```

Comment:
NIT: You're using this string multiple times, across the tests, you could pull it up into a module level constant (something like `CHANGE_DUE_DATE_URL_NAME`) to reduce duplicate and make it easier to change the name in the future.  Find/replace will do the same thing so it's not a huge deal and not something I'd block the PR on but that sort of code cleanliness can make deltas smaller on future changes.